nitransforms/tests/test_surface.py

```diff
@@ -0,0 +1,207 @@
+    assert np.all(scti.reference._triangles == sct.reference._triangles)
+    assert scti == sct
+
+def test_SurfaceCoordinateTransformIO(testdata_path, tmpdir):
+    sphere_reg_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsLR_desc-reg_sphere.surf.gii"
+    pial_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_pial.surf.gii"
+    fslr_sphere_path = testdata_path / "tpl-fsLR_hemi-R_den-32k_sphere.surf.gii"
+
+    sct = SurfaceCoordinateTransform(sphere_reg_path, pial_path)
+    fn = tempfile.mktemp(suffix=".h5")
+    sct.to_filename(fn)
+    sct2 = SurfaceCoordinateTransform.from_filename(fn)
+    assert sct == sct2
+
+def test_ProjectUnproject(testdata_path):
+
+    sphere_reg_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsLR_desc-reg_sphere.surf.gii"
+    fslr_sphere_path = testdata_path / "tpl-fsLR_hemi-R_den-32k_sphere.surf.gii"
+    subj_fsaverage_sphere_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsaverage_desc-reg_sphere.surf.gii"
+    fslr_fsaverage_sphere_path = testdata_path / "tpl-fsLR_space-fsaverage_hemi-R_den-32k_sphere.surf.gii"
+    pial_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_pial.surf.gii"
+
+    # test project-unproject funcitonality
+    projunproj = SurfaceResampler(sphere_reg_path, fslr_sphere_path)
+    with pytest.raises(ValueError):
+        projunproj.apply(pial_path)
+    transformed = projunproj.apply(fslr_fsaverage_sphere_path)
+    projunproj_ref = nb.load(subj_fsaverage_sphere_path)
+    assert (projunproj_ref.agg_data()[0] - transformed._coords).max() < 0.0005
+    assert np.all(transformed._triangles == projunproj_ref.agg_data()[1])
+
+def test_SurfaceResampler(testdata_path, tmpdir):
+    dif_tol = 0.001
+    fslr_sphere_path = testdata_path / "tpl-fsLR_hemi-R_den-32k_sphere.surf.gii"
+    shape_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_thickness.shape.gii"
+    ref_resampled_thickness_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsLR_thickness.shape.gii"
+    pial_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_pial.surf.gii"
+    sphere_reg_path = testdata_path / "sub-sid000005_ses-budapest_acq-MPRAGE_hemi-R_space-fsLR_desc-reg_sphere.surf.gii"
+
+    fslr_sphere = SurfaceMesh(nb.load(fslr_sphere_path))
+    sphere_reg = SurfaceMesh(nb.load(sphere_reg_path))
+    subj_thickness = nb.load(shape_path)
+
+    with pytest.raises(ValueError):
+        SurfaceResampler(sphere_reg_path, pial_path)
+    with pytest.raises(ValueError):
+        SurfaceResampler(pial_path, sphere_reg_path)
+
+    reference = fslr_sphere
+    moving = sphere_reg
+    # compare results to what connectome workbench produces
+    resampling = SurfaceResampler(reference, moving)
+    resampled_thickness = resampling.apply(subj_thickness.agg_data(), normalize='element')
```

Comment:
This brings the cortical thickness from the subject into the fslr sphere, correct?